src/lib/AST.cpp

```diff
@@ -305,26 +318,214 @@ bool TreeAssign::evalBool(std::unordered_map<std::string, variableVal>& vars) co
 
 
 
-void TreeAssign::printInfix() const {
+// Prints variable assignment infix
+void TreeAssign::printInfix(int depth) const {
     if (children.empty()) {
         return;
     }
 
+    for(int i = 1; i <= depth; i++){
+        std::cout << "    ";
+    }
+
     std::cout << '(';
-    children[0]->printInfix();
+    children[0]->printInfix(0);
     for (unsigned int i = 1; i < children.size(); i++) {
         std::cout << " = ";
-        children[i]->printInfix();
+        children[i]->printInfix(0);
     }
     std::cout << ')';
 }
 
 
-
+// Adds a right-most child node
 void TreeAssign::addChild(TreeNode* child){
     children.push_back(child);
 }
 
 
 
-// Incoming logic of TreeStatements
+
+
+
+
+
+
+//
+TreeStatement::TreeStatement(std::string statement){
+    stateStr = statement;
+}
+
+
+
+//
+void TreeStatement::evaluatePrint(std::unordered_map<std::string, variableVal>& vars) const{
+    if(condition->type(vars) == ReturnType::NUM){
+        double value = condition->evalDouble(vars);
+        std::cout << value << '\n';
+        return;
+    }
+
+    if(condition->evalBool(vars)){
+        std::cout << "True\n";
+        return;
+    }
+    std::cout << "False\n";
+    return;
+}
+
+
+
+//
+void TreeStatement::evaluateWhile(std::unordered_map<std::string, variableVal>& vars) const{
+    if(condition->type(vars) != ReturnType::BOOL){
+        throw std::runtime_error("Runtime error: condition is not a bool.");   
+    }
+
+    while(condition->evalBool(vars)){ 
+        for(TreeNode* tree : truths){
+            if(tree->type(vars) == ReturnType::NUM){
+                tree->evalDouble(vars);
+            }
+            else{
+                tree->evalBool(vars);
+            }
+        }
+ 
+        if(condition->type(vars) != ReturnType::BOOL){
+            throw std::runtime_error("Runtime error: condition is not a bool.");   
+        }
+    }
+    return;
+}
+
+
+//
+void TreeStatement::evaluateIf(std::unordered_map<std::string, variableVal>& vars) const{
+    if(condition->type(vars) != ReturnType::BOOL){
+        throw std::runtime_error("Runtime error: condition is not a bool.");   
+    }
+
+    if(condition->evalBool(vars)){
+        for(TreeNode* tree : truths){
+            if(tree->type(vars) == ReturnType::NUM){
+                tree->evalDouble(vars);
+            }
+            else{
+                tree->evalBool(vars);
+            }
+        }
+        return;
+    }
+    
+    for(TreeNode* tree : falses){
+        if(tree->type(vars) == ReturnType::NUM){
+            tree->evalDouble(vars);
+        }
+        else{
+            tree->evalBool(vars);
+        }
+    }
+    return;
+}
+
+
+
+//
```

Comment:
It looks like you are going to do this, but add header comments to describe briefly what these functions do